packages/grafana-ui/src/components/OptionsUI/string.tsx

```diff
@@ -37,3 +43,21 @@ export const StringValueEditor: React.FC<FieldConfigEditorProps<string, StringFi
     />
   );
 };
+
+export const StringWithSuggestionsValueEditor: React.FC<FieldConfigEditorProps<
+  string,
+  StringFieldWithSuggestionsConfigSettings
+>> = ({ value, onChange, item, context }) => {
+  const suggestions = useMemo(() => {
+    return context.getSuggestions ? context.getSuggestions(VariableSuggestionsScope.Values) : [];
+  }, [context.getSuggestions]);
```

Comment:
context.getSuggestions allows consuming the suggestions with the scope of the variables defined. item.settings.getSuggestions is the definition of the suggestion provider. Problem I see here is both of these are available for someone using the API. Any thoughts you have maybe @ryantxu?

Reply:
I don't understand why you need the memo here?  is it so that things are forced to `VariableSuggestionsScope.Values`?

Reply:
It's to make sure `getSuggestions` is not evaluated on every render. `useMemo` memorizes the return value, not the function. But good catch actually, this memoization can lead to weird bugs, as it's not invalidated when i.e. context data change (and can be input for suggestions...)